FeedSX/screens/createPost/index.tsx

```diff
@@ -0,0 +1,549 @@
+          },
+          size: item.size,
+          duration: 0, // You need to specify the duration.
+          pageCount: 0, // You need to specify the page count.
+          url: item.uri,
+        },
+        attachmentType:
+          item.type === 'application/pdf' ? DOCUMENT_ATTACHMENT_TYPE : 0, // You need to specify the attachment type.
+      };
+    });
+    setFormattedDocAttachments([...formattedDocAttachments, ...convertedData]);
+    return convertedData;
+  }
+
+  // convert the LMOGTagsUI to LMAttachmentUI
+  function convertLinkData(data: LMOGTagsUI[]): LMAttachmentUI[] {
+    const convertedData = data.map(item => {
+      return {
+        attachmentMeta: {
+          entityId: '', // You need to specify the entity ID.
+          format: '',
+          name: '',
+          ogTags: {
+            description: item.description,
+            title: item.title,
+            url: item.url,
+            image: item.image,
+          },
+          size: 0,
+          duration: 0, // You need to specify the duration.
+          pageCount: 0, // You need to specify the page count.
+          url: '',
+        },
+        attachmentType: LINK_ATTACHMENT_TYPE, // You need to specify the attachment type.
+      };
+    });
+    setFormattedLinkAttachments([
+      ...formattedLinkAttachments,
+      ...convertedData,
+    ]);
+    return convertedData;
+  }
+
+  //select Images and videoes From Gallery
+  const selectGalley = async (type: string) => {
+    const options = {
+      mediaType: type,
+      selectionLimit: 0,
+    };
+    await launchImageLibrary(options as any, async (response: any) => {
+      if (response?.didCancel) {
+        console.log('didCancel', response.didCancel);
+      }
+      let selectedImages = response?.assets; // selectedImages can be anything images or videos or both
+
+      for (let i = 0; i < selectedImages?.length; i++) {
+        if (selectedImages[i].fileSize >= MAX_FILE_SIZE) {
+          console.log('Files above 100 MB is not allowed');
+          return;
+        }
+      }
+      if (!!selectedImages) {
+        convertData(selectedImages);
+        setShowOptions(false);
+      }
+    });
+  };
+
+  //select Documents From Gallery
+  const selectDoc = async () => {
+    try {
+      const response = await DocumentPicker.pick({
+        type: [DocumentPicker.types.pdf],
+        allowMultiSelection: true,
+      });
+      let selectedDocs: any = response; // selectedImages can be anything images or videos or both
+      let docsArrlength = selectedDocs?.length;
+      if (docsArrlength > 0) {
+        for (let i = 0; i < docsArrlength; i++) {
+          if (selectedDocs[i].size >= MAX_FILE_SIZE) {
+            console.log('Files above 100 MB is not allowed');
+            return;
+          }
+        }
+        if (!!selectedDocs) {
+          convertDocData(selectedDocs);
+          setShowOptions(false);
+        }
+      }
+    } catch (error) {
+      console.log('err', error);
```

Comment:
move this whole code to util file, this is not a view related code.

Reply:
resolved